sky/clouds/service_catalog/common.py

```diff
@@ -530,7 +530,8 @@ def make_list_from_df(rows):
             ),
             axis='columns',
         ).tolist()
-        ret.sort(key=lambda info: (info.accelerator_count, info.cpu_count
+        ret.sort(key=lambda info: (info.accelerator_count, info.cpu_count, info.
+                                   price, info.spot_price
```

Comment:
The `if info.cpu_count is not None else 0` part should follow `info.cpu_count`?

Reply:
That does not seem to be fixing the issue. What is fixing the issue is just removing the cpu_count in the sorting altogether. I don't know if that is viable. I am pushing that now. Lmk if that would work.

Reply:
We probably want to keep the cpu sort key. E.g., it's a bit unintuitive to see some 8-cpu instance in the middle of 4-cpu instances. 
```
» sky show-gpus T4g:1 --all-regions

GPU  QTY  CLOUD  INSTANCE_TYPE  DEVICE_MEM  vCPUs  HOST_MEM  HOURLY_PRICE  HOURLY_SPOT_PRICE  REGION
T4g  1    AWS    g5g.xlarge     16GB        4      8GB       $ 0.420       $ 0.057            us-west-2
T4g  1    AWS    g5g.xlarge     16GB        4      8GB       $ 0.420       $ 0.168            us-east-1
T4g  1    AWS    g5g.xlarge     16GB        4      8GB       $ 0.517       $ 0.055            ap-northeast-2
T4g  1    AWS    g5g.xlarge     16GB        4      8GB       $ 0.525       $ 0.195            eu-central-1
T4g  1    AWS    g5g.2xlarge    16GB        8      16GB      $ 0.556       $ 0.206            us-west-2
T4g  1    AWS    g5g.2xlarge    16GB        8      16GB      $ 0.556       $ 0.242            us-east-1
T4g  1    AWS    g5g.xlarge     16GB        4      8GB       $ 0.567       $ 0.063            ap-northeast-1
T4g  1    AWS    g5g.xlarge     16GB        4      8GB       $ 0.588       $ 0.115            ap-southeast-1
...
```

Reply:
Another problem:
- according to the above, ap-northeast-1 has an on-demand price of $0.567
- this is inconsistent with the following
```
» sky show-gpus T4g:1
*NOTE*: for most GCP accelerators, INSTANCE_TYPE == (attachable) means the host VM's cost is not included.

GPU  QTY  CLOUD  INSTANCE_TYPE  DEVICE_MEM  vCPUs  HOST_MEM  HOURLY_PRICE  HOURLY_SPOT_PRICE  REGION
T4g  1    AWS    g5g.xlarge     16GB        4      8GB       $ 0.420       $ 0.055            ap-northeast-1
...
```

Reply:
okay interesting. Looking into it.

Reply:
Turns out the issue was with the fact that it had a bug to begin with. The value was checked for being None but it was actually never none but nan. So we needed to check for that.

Reply:
Still trying to figure out https://github.com/skypilot-org/skypilot/pull/2933#discussion_r1439684566

Reply:
@concretevitamin for incosistency comes from the fact that we group by `'InstanceType', 'AcceleratorName', 'AcceleratorCount', 'vCPUs', 'MemoryGiB'` when -all-regions is False . And then we call the aggregate with min. This results in taking the min value for each of the remaining column, this is why we always have the ap_northeast-1 region. 

Here: 

```
# Only keep the lowest prices across regions.
            rows = rows.groupby([
                'InstanceType', 'AcceleratorName', 'AcceleratorCount', 'vCPUs',
                'MemoryGiB'
            ],
                                dropna=False).aggregate('min').reset_index()
```

For example when we look at all the rows for  `g5g.xlarge` , then we get: 


```
GPU  QTY  CLOUD  INSTANCE_TYPE  DEVICE_MEM  vCPUs  HOST_MEM  HOURLY_PRICE  HOURLY_SPOT_PRICE  REGION
T4g  1    AWS    g5g.xlarge     16GB        4      8GB       $ 0.420       $ 0.057            us-west-2
T4g  1    AWS    g5g.xlarge     16GB        4      8GB       $ 0.420       $ 0.168            us-east-1
T4g  1    AWS    g5g.xlarge     16GB        4      8GB       $ 0.420       $ 0.178            us-east-1
T4g  1    AWS    g5g.xlarge     16GB        4      8GB       $ 0.420       $ 0.180            us-west-2
T4g  1    AWS    g5g.xlarge     16GB        4      8GB       $ 0.420       $ 0.187            us-east-1
T4g  1    AWS    g5g.xlarge     16GB        4      8GB       $ 0.420       $ 0.189            us-east-1
T4g  1    AWS    g5g.xlarge     16GB        4      8GB       $ 0.420       $ 0.190            us-west-2
T4g  1    AWS    g5g.xlarge     16GB        4      8GB       $ 0.517       $ 0.055            ap-northeast-2
T4g  1    AWS    g5g.xlarge     16GB        4      8GB       $ 0.517       $ 0.074            ap-northeast-2
T4g  1    AWS    g5g.xlarge     16GB        4      8GB       $ 0.525       $ 0.195            eu-central-1
T4g  1    AWS    g5g.xlarge     16GB        4      8GB       $ 0.567       $ 0.063            ap-northeast-1
T4g  1    AWS    g5g.xlarge     16GB        4      8GB       $ 0.588       $ 0.115            ap-southeast-1
T4g  1    AWS    g5g.xlarge     16GB        4      8GB       $ 0.588       $ 0.148            ap-southeast-1
```

And from here we take the minimum of each column to get: 

```
GPU  QTY  CLOUD  INSTANCE_TYPE  DEVICE_MEM  vCPUs  HOST_MEM  HOURLY_PRICE  HOURLY_SPOT_PRICE  REGION
T4g  1    AWS    g5g.xlarge     16GB        4      8GB       $ 0.420       $ 0.055            ap-northeast-1
```

so we get the wrong values. So we need to fix this. Similar issue is happening when I am adding region into the grouping where `HOURLY_PRICE`, and  `HOURLY_SPOT_PRICE`  are independently chosen. 

Am I right to believe that this has been a bug in the code since the start. If it is I am curious, as to how do we want to chose the chepeast option (how to break the tie). I assume we want to sort per region by (HOURLY_PRICE  , HOURLY_SPOT_PRICE ) in that order and chose the first element. 

Let me know if that is a fair assumption

Reply:
Good catch! Yes, it's a problem on master too.